pkg/config/exec_config.go

```diff
@@ -0,0 +1,125 @@
+package config
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/warptools/warpforge/pkg/workspace"
+)
+
+// ExecConfig is warpforge's internal configuration required to run Exec
+type ExecConfig struct {
+	// binPath is the path containing required binaries to run (rio, runc)
+	binPath string
+	// keepRunDir can be set to prevent removing the created temporary directory after execution
+	keepRunDir bool
+	// runPath is the path used to store temporary files used for formula run
+	runPath Override[string]
+	// warehousePath may be set to override the warehouse path used for... TODO
+	warehousePath Override[string]
+	// WorkingDirectory should be set to the result of os.Getwd
+	workingDirectory string
+	// WorkspaceSet is the workspace stack set to be used when executing.
+	// This controls behavior such as memoization and host warehouses.
+	workspaceSet workspace.WorkspaceSet
+}
+
+func (e ExecConfig) BinPath() string {
+	return e.binPath
+}
+
+func (e ExecConfig) KeepRunDir() bool {
+	return envIsSet(EnvWarpforgeKeepRundir)
+}
+
+func (e ExecConfig) RunPathOverride() (string, bool) {
+	return e.runPath.get()
+}
+
+func (e ExecConfig) WarehousePathOverride() (string, bool) {
+	return e.warehousePath.get()
+}
+
+func (e ExecConfig) WorkingDirectory() string {
+	return e.workingDirectory
+}
+func (e ExecConfig) WorkspaceSet() workspace.WorkspaceSet {
+	return e.workspaceSet
+}
+
+// NewConfig returns a warpforge configuration
+//
+// Errors:
+//
+//   - warpforge-error-initialization -- unable to load value
+//   - warpforge-error-searching-filesystem -- unable to load workspace stack
+func NewConfig() (ExecConfig, error) {
```

Comment:
Would like to have this function take more parameters.  (I know this PR is an incremental improvement on gathering a bunch of this stuff from happening in many places to just one, but I'd also kinda like to take this one step further while we're here, and have less ambient magic happening too.)

Seems like the fsys and cwd as a parameter, are the two biggest ones.  A `getEnv func(string) string` or `envOverride map[string]string` might be a good idea too.

Or: if the parameter list seems like it's then getting too long and unwieldy: okay, fine; then let's put them in another struct, and have some sort of Init function on it that does the ambient-read stuff (e.g. all the Getwd and GetEnv stuff).

Any design that has an entrypoint that's got no ambient reads is a thumbs up for me.  It can have a zero-arg super-magic function that does all the ambient reads too; I just don't want to be stuck with that as the only entrypoint.

Reply:
Adding an `FS() fs.FS` might be a good idea. But consumers can choose what parts of the interface they want or implement new ones, so overrides feel somewhat redundant. I'd want an example of say `formulaexec.Exec` _taking_ an interface with an `FS()` method and using it before implementing it.